### PR TITLE
Add test coverage for coverage.rb

### PR DESCRIPTION
**Pull request by smridge**
Adds _some_ test coverage to main area of attraction with light refactoring.

Although there is healthy debate around testing private methods, `collect_coverage` is the main source of things. I wanted to easily test the counts etc. This is a bit difficult with `report` returning an `exit` code always.

## TODO
- Finish scenarios for `collect_coverage`

### CODE REVIEW

File: lib/swagcov/coverage.rb [hunk 1]
```diff
@@ -13,20 +15,25 @@ def initialize
...
+      final_output
+
+      exit @total - @covered
+    end
+
+    private
+
+    attr_reader :dotfile
 
+    def collect_coverage
```
Comment (lonelyelk) [inline lib/swagcov/coverage.rb:32]: I think that the main debate around testing private method is not exactly our kind of debate. When trying to test this what you encounter is more of a design problem. I think there will be less of dilemmas if we were to split this class in two. Or even three. Because this class is doing at least two things. You tried making one of those things private but it's the one that requires more testing.

If I were to really force that «S» in S.O.L.I.D., I'd suggest to have `AppRoutesProcessor`, `CoverageCollector` and `CoverageReporter`. (Maybe more human and less Java™ Enterprise names would be better.) In that case most of the edge cases could be tested more thoroughly in isolation. And you only need a few integration tests.

If you liked what extracting `Dotfile` did to the application, I think you might like pulling those parts out too.

**Review by lonelyelk — APPROVED** [review]
I like that test coverage is improved. I also think it's a good start for the refactoring. In my experience a lot of design improvements come from trying to test an application or improve the testability.

File: lib/swagcov/coverage.rb (hunk 1 above)
Comment (smridge) [reply lib/swagcov/coverage.rb:32]: I agree 💯 . I definitely realized this when writing tests 🙃
